test/e2e/deployment.go

```diff
@@ -317,3 +322,71 @@ func testRolloverDeployment(f *Framework) {
 	newRC, err = deploymentutil.GetNewRC(*deployment, c)
 	Expect(newRC.Spec.Template.Spec.Containers[0].Image).Should(Equal(updatedDeploymentImage))
 }
+
+func testPausedDeployment(f *Framework) {
+	ns := f.Namespace.Name
+	c := f.Client
+	deploymentName := "nginx"
+	podLabels := map[string]string{"name": "nginx"}
+	Logf("Creating paused deployment %s", deploymentName)
+	_, err := c.Deployments(ns).Create(&extensions.Deployment{
+		ObjectMeta: api.ObjectMeta{
+			Name: deploymentName,
+		},
+		Spec: extensions.DeploymentSpec{
+			Replicas:       1,
+			Selector:       podLabels,
+			UniqueLabelKey: extensions.DefaultDeploymentUniqueLabelKey,
+			Paused:         true,
+			Template: api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: podLabels,
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "nginx",
+							Image: "nginx",
+						},
+					},
+				},
+			},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		deployment, err := c.Deployments(ns).Get(deploymentName)
+		Expect(err).NotTo(HaveOccurred())
+		Logf("deleting deployment %s", deploymentName)
+		Expect(c.Deployments(ns).Delete(deploymentName, nil)).NotTo(HaveOccurred())
+		// TODO: remove this once we can delete rcs with deployment
+		newRC, err := deploymentutil.GetNewRC(*deployment, c)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(c.ReplicationControllers(ns).Delete(newRC.Name)).NotTo(HaveOccurred())
+	}()
+	// Check that deployment is created fine.
+	deployment, err := c.Deployments(ns).Get(deploymentName)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Verify that there are no realized states for the paused deployment.
+	rcs, err := deploymentutil.GetRCsForDeployment(c, *deployment)
```

Comment:
I prefer checking for the actual rc rather than an event. Let's get the way we get the rc sorted out (GetRCsForDeployment vs GetOldRCs/GetNewRC.